lib/scholar/metrics/classification.ex

```diff
@@ -672,11 +674,70 @@ defmodule Scholar.Metrics.Classification do
         {true_positive, false_positive, false_negative}
 
       _ ->
-      {true_positive, false_positive, false_negative}
+        {true_positive, false_positive, false_negative}
     end
   end
 
-  defnp precision_recall_fscore_n(y_true, y_pred, beta, num_classes, average) do
+  @doc """
+  Calculates precision, recall, F-score and support for each
+  class. It also supports a `beta` argument which weights
+  recall more than precision by it's value.
+
+  ## Options
+
+  #{NimbleOptions.docs(@precision_recall_fscore_support_schema)}
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([0, 1, 1, 1, 1, 0, 2, 1, 0, 1], type: :u32)
+      iex> y_pred = Nx.tensor([0, 2, 1, 1, 2, 2, 2, 0, 0, 1], type: :u32)
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3)
+      #Nx.Tensor<
+        f32[3]
+        [0.6666666865348816, 0.6666666865348816, 0.4000000059604645]
+      >
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3, average: :macro)
+      #Nx.Tensor<
+        f32
+        0.5777778029441833
+      >
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3, average: :weighted)
+      #Nx.Tensor<
+        f32
+        0.6399999856948853
+      >
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(y_true, y_pred, num_classes: 3, average: :micro)
+      #Nx.Tensor<
+        f32
+        0.6000000238418579
+      >
+      iex> Scholar.Metrics.Classification.precision_recall_fscore_support(Nx.tensor([1, 0, 1, 0]), Nx.tensor([0, 1, 0, 1]), 2, num_classes: 2, average: :none)
+      #Nx.Tensor<
+        f32[2]
+        [0.0, 0.0]
+      >
+  """
+  deftransform precision_recall_fscore_support(y_true, y_pred, beta \\ 1, opts \\ [])
```

Comment:
Is there a better way to have this possibility of having the `beta` value as default 1?

Reply:
Yes, there is a better way. If you want to have a default on a value that needs to be passed, then use NimbleOptions schema with `default: 1`.

Reply:
Like [here](https://github.com/elixir-nx/scholar/blob/93d3da6bce3d47b44f5c30b7ac9c20aaa7984f00/lib/scholar/metrics/classification.ex#L81)

Reply:
Should the arguments default to use the NimbleOptions schema for these kinds of default arguments? I was thinking of having the API follow the same way as in `fbeta_score`. If we move to use NimbleOptions, I think that function should also change (the `beta` argument), so it's "consistent" with each other. What do you think?

Reply:
Yes, I think it's a good idea to change `fbeta_score` to accept `:beta` parameter via `opts`. In the whole `Scholar` it's a standard to pass even very important params through opts like `k` in `K-means`